src/shared_modules/utils/archiveHelper.hpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Utils abstract wait
+ * Copyright (C) 2015, Wazuh Inc.
+ * Feb 8, 2024.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _ARCHIVE_HELPER_HPP
+#define _ARCHIVE_HELPER_HPP
+
+#include "archive.h"
+#include "archive_entry.h"
+#include "customDeleter.hpp"
+#include <filesystem>
+
+template<typename F, typename G, F func1, G func2>
+struct Deleter
+{
+    template<typename T>
+    constexpr void operator()(T* arg) const
+    {
+        func1(arg);
+        func2(arg);
+    }
+};
+
+// C++ smart pointers usage to avoid manually free allocated memory.
+using ArchiveReadPtr = std::unique_ptr<
+    struct archive,
+    Deleter<decltype(&archive_read_close), decltype(&archive_read_free), archive_read_close, archive_read_free>>;
+using ArchiveWritePtr = std::unique_ptr<
+    struct archive,
+    Deleter<decltype(&archive_write_close), decltype(&archive_write_free), archive_write_close, archive_write_free>>;
+
+namespace Utils
+{
+    /**
+     * @brief Decompression for .tar compressed files.
+     *
+     */
+    class ArchiveHelper
+    {
+    private:
+        static void copyData(struct archive* archiveRead, struct archive* archiveWrite)
+        {
+            int retVal;
+            const void* buff;
+            size_t size;
+            int64_t offset;
+
+            for (;;)
+            {
+                retVal = archive_read_data_block(archiveRead, &buff, &size, &offset);
+                if (retVal == ARCHIVE_EOF)
+                {
+                    break;
+                }
+                else if (retVal != ARCHIVE_OK)
+                {
+                    throw std::runtime_error("Couldn't read file");
+                }
```

Comment:
if else-if needs else.

Reply:
I reverted the use of else-if to use only if statements as it was initially.

Reply:
Is that true for this scenario?
```cpp
                if (retVal == ARCHIVE_EOF)
                {
                    break;
                }
                else if (retVal != ARCHIVE_OK)
                {
                    throw std::runtime_error("Couldn't read file");
                }
```
                
The last possible state of retVal at this is ARCHIVE_OK. Adding another seems redundant

Reply:
LGTM, the change.